### PR TITLE
Add Leres depth estimator

### PR DESCRIPTION
**Pull request by pdoane**

### CODE REVIEW

File: src/controlnet_aux/leres/__init__.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        pix2pixmodel.save_dir = os.path.dirname(model_path)
+        pix2pixmodel.load_networks('latest')
+        pix2pixmodel.eval()
+
+        return cls(model, pix2pixmodel)
+
+    def to(self, device):
+        self.model.to(device)
+        # TODO - refactor pix2pix implementation to support device migration
+        # self.pix2pixmodel.to(device)
```
Comment (pdoane) [inline src/controlnet_aux/leres/__init__.py:56]: Upstream version has not been refactored for supporting non-cuda devices - we can do that in a separate PR or wait and see if it gets addressed.

File: src/controlnet_aux/leres/pix2pix/util/get_data.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+from __future__ import print_function
```
Comment (patrickvonplaten) [inline src/controlnet_aux/leres/pix2pix/util/get_data.py:1]: Do we need this file? There are some packages that we don't install by default like `bs4`

File: src/controlnet_aux/leres/pix2pix/util/html.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import dominate
```
Comment (patrickvonplaten) [inline src/controlnet_aux/leres/pix2pix/util/html.py:1]: do we need this?

File: src/controlnet_aux/leres/pix2pix/util/visualizer.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import os
+import sys
+import ntpath
```
Comment (patrickvonplaten) [inline src/controlnet_aux/leres/pix2pix/util/visualizer.py:4]: is this a package?

**Review by patrickvonplaten — COMMENTED** [review]
In general, this is good to go for me - thanks a lot! Just there are a couple of pypi packages that are added - do we need all of these files (beautiful soup a.k.a `bs4` e.g. or `dominate` => could we maybe remove those files?)

**Comment by pdoane** [comment]
Pruned util directory to what is needed.  @patrickvonplaten - should be good to go now.

**Comment by patrickvonplaten** [comment]
Super! Good to merge for me!
